core/Controller/TaskProcessingApiController.php

```diff
@@ -0,0 +1,357 @@
+				'tasks' => $json,
+			]);
+		} catch (Exception $e) {
+			return new DataResponse(['message' => $this->l->t('Internal error')], Http::STATUS_INTERNAL_SERVER_ERROR);
+		} catch (\JsonException $e) {
+			return new DataResponse(['message' => $this->l->t('Internal error')], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	/**
+	 * This endpoint returns the contents of a file referenced in a task
+	 *
+	 * @param int $taskId The id of the task
+	 * @param int $fileId The file id of the file to retrieve
+	 * @return DataDownloadResponse<Http::STATUS_OK, string, array{}>|DataResponse<Http::STATUS_INTERNAL_SERVER_ERROR|Http::STATUS_NOT_FOUND, array{message: string}, array{}>
+	 *
+	 *  200: File content returned
+	 *  404: Task or file not found
+	 */
+	#[NoAdminRequired]
+	#[ApiRoute(verb: 'GET', url: '/tasks/{taskId}/file/{fileId}', root: '/taskprocessing')]
+	public function getFileContents(int $taskId, int $fileId): Http\DataDownloadResponse|DataResponse {
+		try {
+			$task = $this->taskProcessingManager->getUserTask($taskId, $this->userId);
+			$ids = $this->extractFileIdsFromTask($task);
+			if (!in_array($fileId, $ids)) {
+				return new DataResponse(['message' => $this->l->t('Not found')], Http::STATUS_NOT_FOUND);
+			}
+			$node = $this->rootFolder->getFirstNodeById($fileId);
+			if ($node === null) {
+				$node = $this->rootFolder->getFirstNodeByIdInPath($fileId, '/' . $this->rootFolder->getAppDataDirectoryName() . '/');
+				if (!$node instanceof File) {
+					throw new \OCP\TaskProcessing\Exception\NotFoundException('Node is not a file');
+				}
+			} elseif (!$node instanceof File) {
+				throw new \OCP\TaskProcessing\Exception\NotFoundException('Node is not a file');
+			}
+			return new Http\DataDownloadResponse($node->getContent(), $node->getName(), $node->getMimeType());
+		} catch (\OCP\TaskProcessing\Exception\NotFoundException $e) {
+			return new DataResponse(['message' => $this->l->t('Not found')], Http::STATUS_NOT_FOUND);
+		} catch (GenericFileException|NotPermittedException|LockedException|Exception $e) {
+			return new DataResponse(['message' => $this->l->t('Internal error')], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	/**
+	 * @param Task $task
+	 * @return list<mixed>
+	 * @throws \OCP\TaskProcessing\Exception\NotFoundException
+	 */
+	private function extractFileIdsFromTask(Task $task) {
+		$ids = [];
+		$taskTypes = $this->taskProcessingManager->getAvailableTaskTypes();
+		if (!isset($taskTypes[$task->getTaskTypeId()])) {
+			throw new \OCP\TaskProcessing\Exception\NotFoundException('Could not find task type');
+		}
+		$taskType = $taskTypes[$task->getTaskTypeId()];
+		foreach ($taskType['inputShape'] + $taskType['optionalInputShape'] as $key => $descriptor) {
+			if (in_array(EShapeType::getScalarType($descriptor->getShapeType()), [EShapeType::File, EShapeType::Image, EShapeType::Audio, EShapeType::Video], true)) {
+				if (is_array($task->getInput()[$key])) {
+					$ids += $task->getInput()[$key];
```

Comment:
```suggestion
				$value = $task->getInput()[$key];
				if (is_array($value)) {
					$ids += $value;
```
Avoid calling the method several times.